dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp

```diff
@@ -1074,9 +1153,62 @@ TEST(DataInterfaceIIDMTest, testHvdcLineVscConvertersIIDM) {
   ASSERT_TRUE(bus->hasConnection());
   hvdc->hasDynamicModel(false);
   ASSERT_TRUE(bus->hasConnection());
-}  // TEST(DataInterfaceIIDMTest, testHvdcLineVscConvertersIIDM)
+}  // TEST(DataInterfaceIIDMTest, testConnectedHvdcLineVscConvertersIIDM)
+
+TEST(DataInterfaceIIDMTest, testDisconnectedHvdcLineVscConvertersIIDM) {
+  const BusBreakerNetworkProperty properties = {
+      false /*instantiateCapacitorShuntCompensator*/,
+      false /*instantiateStaticVarCompensator*/,
+      false /*instantiateTwoWindingTransformer*/,
+      false /*instantiateRatioTapChanger*/,
+      false /*instantiatePhaseTapChanger*/,
+      false /*instantiateDanglingLine*/,
+      false /*instantiateGenerator*/,
+      false /*instantiateLccConverterWithConnectedHvdc*/,
+      false /*instantiateLccConverterWithDisconnectedHvdc*/,
+      false /*instantiateLine*/,
+      false /*instantiateLoad*/,
+      false /*instantiateSwitch*/,
+      false /*instantiateVscConverterWithConnectedHvdc*/,
+      true /*instantiateVscConverterWithDisconnectedHvdc*/,
+      false /*instantiateThreeWindingTransformer*/,
+      false /*instantiateBattery*/
+  };
+  shared_ptr<DataInterfaceIIDM> data = createDataItfFromNetwork(createBusBreakerNetwork(properties));
+  exportStateVariables(data);
+  boost::shared_ptr<HvdcLineInterfaceIIDM> hvdc = boost::dynamic_pointer_cast<HvdcLineInterfaceIIDM>(data->findComponent("MyHvdcLine"));
+  boost::shared_ptr<VscConverterInterfaceIIDM> vsc1 = boost::dynamic_pointer_cast<VscConverterInterfaceIIDM>(hvdc->getConverter1());
+  boost::shared_ptr<VscConverterInterfaceIIDM> vsc2 = boost::dynamic_pointer_cast<VscConverterInterfaceIIDM>(hvdc->getConverter2());
+
+  ASSERT_TRUE(std::isnan(vsc1->getVscIIDM().getTerminal().getP()));
```

Comment:
Would it be possible to add a test in which the NaN stays a NaN?

Reply:
Yes it is. I've just added this in 2 new commits.